### ISSUE TITLE
Using a different input tensor shape in a Vgg model

### ISSUE THREAD

**Issue by paulbonnard** [keras-team/keras#12609]
Hello, 

i'm working on imagery with multiple channels. I would like to use a vgg model as  part of a bigger gan model.

I would like to use for example tensor with shape (400,400,4) but vgg model only accept shape (400,400,3)

So i look at the keras documentation, it tolds that you can change the input layer with  the command include_top = False and add a new layer as input but it doesn't work.

So i thought to add a new convolutionnal layer at the beggining of the vgg model with 3 filters as output but it doesn't work either ?

Did you already have a problem like mine ? 

here is my code
`from keras import Input
from keras.applications import VGG19
from keras.layers.convolutional import Conv2D
from keras.models import Model

sh = (400,400,4)

input_layer = Input(shape = sh)
conv = Conv2D(filters=4, kernel_size=3, strides=2, padding='same')(input_layer)

v =  VGG19(weights="imagenet", include_top = False ,input_tensor =  conv)
output_layer = v.output

model = Model(inputs=[input_layer], outputs=[output_layer])`

**Comment by kashyapraval** [comment]
Having a 4th frame input leads to differences that cascade down and the trained weights in VGG may not remain as effective. However, if you intend to train the complete model by yourself, then you can take inspiration from the model architecture.

**Comment by paulbonnard** [comment]
Yes, this what i was thinking about too.
Idealy, i would like to not train a the vgg model.

My images are pretty différent than the ones you find in imagenet.

Actually i was just for do a test.

But what do you think about my problem, i havent the label to train a vgg on my datas. 

However taking différent label to train the vgg with my 4 channels datas is not very appropriate

**Comment by kashyapraval** [comment]
In the absence of labels, training is out of the question. If you intend to tag your images using VGG, and if you have the color space as R-G-B-A(alpha), you could ignore the 4th frame and use the 3 frames. In case of CMYK as well, you could convert the same to RGB.

**Comment by paulbonnard** [comment]
OK thank you
